helpers/spark-defaults.conf

```diff
@@ -0,0 +1,7 @@
+spark.app.name=canonical-spark-default-app
+spark.executor.instances=2
+spark.kubernetes.container.image=docker.io/averma32/sparkpy6:latest
+spark.kubernetes.container.image.pullPolicy=Always
+spark.kubernetes.namespace=default
+spark.kubernetes.authenticate.driver.serviceAccountName=spark
```

Comment:
it might make sense to generate this file rather than using static content, since I think the service account name can be configured by the helper script IIRC

Reply:
I'd just like to clarify the user-experience we are imaging before having @averma-canonical addressing this, so to have clear requirements when it comes configuration centralization and handling.

After reading the discussion, and some thoughts, I'd imagine the following process:

(foreword) The defaults for spark-submit have **two possible sources**: sensible configuration we package within the SNAP (e.g. spark.kubernetes.container.image) and installation configuration (e.g. spark.kubernetes.namespace). 

1. The first (packaged configuration) are stored in SNAP read-only location
2. The second (installation configuration) can be generated dynamically (as suggested by @grobbie) and merged with the packaged configuration during setup. 
3. The setup writes the merged file to a writable location (either within SNAP or /etc like location) and this could also be edited by the user, if the user feels these modifications will generally propagate to most of his/her jobs
4. Spark-submit/shell/pyspark use the merged (and eventually user customized) configuration file, using --conf flags for particular custom job configurations.

Hope the outlined process is clear. But If not, we could have a call for discussing and reviewing this.

Reply:
After talking to both @grobbie and @delgod , we agreed on the configuration handling outlined [here](https://docs.google.com/presentation/d/1I3htuJEoV4rVAVdQT6dL4bh5vGtYbvRN9tpz3AAxZhM), which will be implemented step-by-step (as outlined in the doc). This PR deals with the first task